metrics/prometheus.go

```diff
@@ -0,0 +1,183 @@
+package metrics
+
+import (
+	"net/http"
+	"os"
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	namespace = "traefik"
+)
+
+// TODO: #156 - More detailed metrics
+type entrypointMetrics struct {
+}
+
+// TODO: #156 - More detailed metrics
+type providerMetrics struct {
+}
+
+// TODO: #156 - More detailed metrics
+type frontendMetrics struct {
+}
+
+// TODO: #156 - More detailed metrics
+type backendMetrics struct {
+}
+
+type globalMetrics struct {
+	UpTime                    prometheus.Gauge
```

Comment:
The process collector will take care of this for you. It exposes the timestamp the process started at, which is more reliable and easier to work with than an uptime.